pkg/pi/pi.go

```diff
@@ -35,6 +37,10 @@ func NewPi(cfg *config.Config) *Pi {
 	p.openEtcd()
 	p.watchGlobalCfg()
 
+	if err := agent.Listen(agent.Options{}); err != nil {
```

Comment:
// ShutdownCleanup automatically cleans up resources if the
	// running process receives an interrupt. Otherwise, users
	// can call Close before shutting down.
	// Optional.
	ShutdownCleanup bool

May be this option is useful

agent.Options{
		ShutdownCleanup: true, // automatically closes on os.Interrupt
	}

Reply:
done